notebooks/speculative-sampling/speculative-sampling.ipynb

```diff
@@ -75,9 +75,17 @@
    "metadata": {
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #11.    for i, prompt in enumerate(prompts):
</pre><p>with increasing number of examples this log may be too long, maybe better to use progress bar (using tqdm) instead of printing number of executed example?</p><p> </p><p>printing raw statistic is not very informative, maybe better aggregate them? e.g. print only mean inference time?</p><p> </p><p>totals AR and totals is not very descreptive names (for me AR usually means "action required" and google also gives me at least 5 different abbreviations including Arcansas, augmented reality and some chemistry and medical termins :)  if you want to use it I think it should be used somewhere in description  in braсkets (Auto-regressive (AR))</p><p> </p><p>Also it will be good to add subtitle for that for clirance in notebook structure</p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/2609/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Reply:
added progress bar, and re-naming.
Regarding raw statistics - not sure it make sense to aggregate since time per example may differ significantly. The best option might be to not printing it, and leave it as debug option, do you agree?

Reply:
yes, not printing is also fine for me as anyway, you have this comparison in next cell.
Also, I am not sure, but possibly representing it on chart (putting both auto-regressive and speculative decoding for each sample together) for visual comparison may be more effective, but it is not obligated to have it